### ISSUE TITLE
Error when pasting long code containing multibyte characters into the radian console and executing it

### ISSUE THREAD

**Issue by eitsupi** [randy3k/radian#377]
When I paste a long code of multibyte characters into the radian console and try to run it, I get the following error and cannot run it.
This does not occur with the default R terminal.

```
Error: EOF whilst reading MBCS char at line 5
Error: invalid multibyte character in parser at line 1
```

The following code reproduces this in my environment.
(The following strings are randomly generated.)

```r
a <- c("おゎがごなめかぅさんんぴしもあみぺえぅぞかおもづどいたぼぺぁりうもぞさせめにげさほぅぅもあぶまゆぜゎまやつりゅびけぁむれぴりゃぎぴせぐゑきぢぷぅるおみらへつにすぅょよへょこぺぁひおべいけとみぺととびばぶゐぺぴのるぃゅぽぐやじぶみあいさざゔぅまだおゎがごなめかぅさんんぴしもあみぺえぅぞかおもづどいたぼぺぁりうもぞさせめにげさほぅぅもあぶまゆぜゎまやつりゅびけぁむれぴりゃぎぴせぐゑきぢぷぅるおみらへつにすぅょよへょこぺぁひおべいけとゎべぢでみぷゃぜてこごねぼぶづなぺれぶゑげらぷぼだぱんかこべとこるぐゔなへさばろすてへわたなさゐぜぺてぉでてずけゆぼれきゅはむぇだぴづにをぉわえじごふちりっぅぷあぜぷやべろうまぞてすやなげぅずでょぱけんぬくやわもぱぼゑどぱげをらひゕぜこ
おゎがごなめかぅさんんぴしもあみぺえぅぞかおもづどいたぼぺぁりうもぞさせめにげさほぅぅもあぶまゆぜゎまやつりゅびけぁむれぴりゃぎぴせぐゑきぢぷぅるおみらへつにすぅょよへょこぺぁひおべいけとみぺととびばぶゐぺぴのるぃゅぽぐやじぶみあいさざゔぅまだおゎがごなめかぅさんんぴしもあみぺえぅぞかおもづどいたぼぺぁりうもぞさせめにげさほぅぅもあぶまゆぜゎまやつりゅびけぁむれぴりゃぎぴせぐゑきぢぷぅるおみらへつにすぅょよへょこぺぁひおべいけとゎべぢでみぷゃぜてこごねぼぶづなぺれぶゑげらぷぼだぱんかこべとこるぐゔなへさばろすてへわたなさゐぜぺてぉでてずけゆぼれきゅはむぇだぴづにをぉわえじごふちりっぅぷあぜぷやべろうまぞてすやなげぅずでょぱけんぬくやわもぱぼゑどぱげをらひゕぜこ
おゎがごなめかぅさんんぴしもあみぺえぅぞかおもづどいたぼぺぁりうもぞさせめにげさほぅぅもあぶまゆぜゎまやつりゅびけぁむれぴりゃぎぴせぐゑきぢぷぅるおみらへつにすぅょよへょこぺぁひおべいけとみぺととびばぶゐぺぴのるぃゅぽぐやじぶみあいさざゔぅまだおゎがごなめかぅさんんぴしもあみぺえぅぞかおもづどいたぼぺぁりうもぞさせめにげさほぅぅもあぶまゆぜゎまやつりゅびけぁむれぴりゃぎぴせぐゑきぢぷぅるおみらへつにすぅょよへょこぺぁひおべいけとゎべぢでみぷゃぜてこごねぼぶづなぺれぶゑげらぷぼだぱんかこべとこるぐゔなへさばろすてへわたなさゐぜぺてぉでてずけゆぼれきゅはむぇだぴづにをぉわえじごふちりっぅぷあぜぷやべろうまぞてすやなげぅずでょぱけんぬくやわもぱぼゑどぱげをらひゕぜこ
おゎがごなめかぅさんんぴしもあみぺえぅぞかおもづどいたぼぺぁりうもぞさせめにげさほぅぅもあぶまゆぜゎまやつりゅびけぁむれぴりゃぎぴせぐゑきぢぷぅるおみらへつにすぅょよへょこぺぁひおべいけとみぺととびばぶゐぺぴのるぃゅぽぐやじぶみあいさざゔぅまだおゎがごなめかぅさんんぴしもあみぺえぅぞかおもづどいたぼぺぁりうもぞさせめにげさほぅぅもあぶまゆぜゎまやつりゅびけぁむれぴりゃぎぴせぐゑきぢぷぅるおみらへつにすぅょよへょこぺぁひおべいけとゎべぢでみぷゃぜてこごねぼぶづなぺれぶゑげらぷぼだぱんかこべとこるぐゔなへさばろすてへわたなさゐぜぺてぉでてずけゆぼれきゅはむぇだぴづにをぉわえじごふちりっぅぷあぜぷやべろうまぞてすやなげぅずでょぱけんぬくやわもぱぼゑどぱげをらひゕぜこ
おゎがごなめかぅさんんぴしもあみぺえぅぞかおもづどいたぼぺぁりうもぞさせめにげさほぅぅもあぶまゆぜゎまやつりゅびけぁむれぴりゃぎぴせぐゑきぢぷぅるおみらへつにすぅょよへょこぺぁひおべいけとみぺととびばぶゐぺぴのるぃゅぽぐやじぶみあいさざゔぅまだおゎがごなめかぅさんんぴしもあみぺえぅぞかおもづどいたぼぺぁりうもぞさせめにげさほぅぅもあぶまゆぜゎまやつりゅびけぁむれぴりゃぎぴせぐゑきぢぷぅるおみらへつにすぅょよへょこぺぁひおべいけとゎべぢでみぷゃぜてこごねぼぶづなぺれぶゑげらぷぼだぱんかこべとこるぐゔなへさばろすてへわたなさゐぜぺてぉでてずけゆぼれきゅはむぇだぴづにをぉわえじごふちりっぅぷあぜぷやべろうまぞてすやなげぅずでょぱけんぬくやわもぱぼゑどぱげをらひゕぜこ")
```

**Comment by randy3k** [comment]
I think I understand what it fails. It is basically because we need to break a long text into chunks of exact size of 4094 bytes. However, for multibytes characters, it is obviously problematic because we might cut the character to halves. Base R works around it by breaking text at line breaks.

However, radian doesn't simply work by breaking long text at new lines because radian parses the whole long text at once. I will need to rework on the parsing logic.

Here are two workarounds for this situation.

1. replace newline in string by `\n`. i.e. instead of
```
a <- "abc
edf"
```
use
```
a <- "abc\nedf"
```
However, it will probably break base R.

2. To support both base R and radian, a better solution is to save the string in a `.R` file and source the R file.

**Comment by eitsupi** [comment]
Thank you for digging into this!
At the moment I get around this by using the default R console for projects that need to deal with long strings, but not being able to use radian is stressful.

**Comment by randy3k** [comment]
Oh, I was actually wrong. It is due to a bug in [rchitect](https://github.com/randy3k/rchitect).
Installing from source should fix the issue.
```
pip install -U git+https://github.com/randy3k/rchitect
```
A newer version has been tagged and will be released soon.

**Comment by randy3k** [comment]
I have created a PR trying to fix this: #379 

it would be great if you could help testing it.

**Comment by eitsupi** [comment]
Thank you for the quick fixing.
I tried the latest version (`python3 -m pip install git+https://github.com/randy3k/radian.git@refs/pull/379/head`) and it seems working fine for me!
